packages/md-enhance/src/client/components/playground/PlaygroundExternal.ts

```diff
@@ -0,0 +1,97 @@
+import { defineComponent, h, computed, ref, Ref } from "vue";
+
+import type { VNode } from "vue";
+
+import "../../styles/playground.scss";
+import { LOADING_SVG, PLAY_SVG } from "../icons";
+import { usePlaygroundExternal } from "../../composables/playground";
+import { parsePlaygroundSettings } from "../../utils/playground";
+
+export default defineComponent({
+  name: "PlaygroundExternal",
+
+  props: {
+    title: { type: String, default: "" },
+    config: { type: String, required: true },
+    settings: { type: String, default: "{}" },
+    id: { type: String, required: true },
+  },
+
+  setup(props) {
+    const playgroundContainer = ref<HTMLElement | null>(null);
+    const iframe = ref<HTMLElement | null>(null);
+
+    const loading = ref<boolean>(true);
+
+    const playgroundOptions = parsePlaygroundSettings(props.settings);
+
+    const previewLink: Ref<string> = computed(() => {
+      const { link } = usePlaygroundExternal(
+        props.config,
+        playgroundOptions.external || {}
+      );
+
+      return link;
+    });
+
+    const hideLoading = () => {
```

Comment:
If a function is kind of simple like this, and we are not using it in multiple places, we may prefer to write the function directly in render function